app/src/main/java/ca/marshallasch/veil/DataStore.java

```diff
@@ -0,0 +1,198 @@
+            try {
+                postPair = hashTableStore.findPostByHash(hash);
+            }
+            catch (TooManyResultsException e) {
+                e.printStackTrace();
+            }
+
+            // add the post to the list
+            if (postPair != null) {
+                posts.add(postPair);
+            }
+        }
+
+        return posts;
+    }
+
+    /**
+     * Gets the list of comments for a post sorted in by time posted for a given post.
+     * @param postHash the hash of the post that the comments will be found for.
+     * @return a list of comment pairs
+     */
+    public List<Pair<String, DhtProto.Comment>> getCommentsForPost(String postHash) {
+
+        List<String> hashes = db.getCommentHashes(postHash);
+
+        Pair<String, DhtProto.Comment> commentPair;
+        List<Pair<String, DhtProto.Comment>> comments = new ArrayList<>();
+
+        // get each post that is in the list
+        for (String hash: hashes) {
+
+            commentPair = null;
+            try {
+                commentPair = hashTableStore.findCommentByHash(hash);
+            }
+            catch (TooManyResultsException e) {
+                e.printStackTrace();
+            }
+
+            // add the post to the list
+            if (commentPair != null) {
+                comments.add(commentPair);
+            }
+        }
+
+        Collections.sort(comments, new CommentPairComparator());
+
+        return comments;
+
+    }
+
+
+    public Sync.MappingMessage getDatabase() {
+
+        List<Pair<String, String>> knownPosts = db.dumpKnownPosts();
+
+        Sync.MappingMessage.Builder builder = Sync.MappingMessage.newBuilder();
+
+        // add it to the list
+        for (Pair<String, String> pair: knownPosts) {
+            builder.addMappings(Sync.CommentMapping.newBuilder()
+                    .setPostHash(pair.first)
+                    .setCommentHash(pair.second)
+                    .build());
+        }
+
+        // build the message to send to other devices
+        return builder.build();
+    }
+
+    public Sync.HashData getDataStore() {
+
+        List<Pair<String, DhtProto.DhtWrapper>> data = hashTableStore.getData();
+
+
+        Sync.HashData.Builder builder = Sync.HashData.newBuilder();
+
+        // generate the list
+        for(Pair<String, DhtProto.DhtWrapper> pair: data) {
+
+            builder.addEntries(Sync.HashPair.newBuilder()
+                    .setHash(pair.first)
+                    .setEntry(pair.second)
+                    .build());
+        }
+
+        return builder.build();
+    }
+
+    public void syncDatabase(Sync.MappingMessage message) {
+
+        List<Sync.CommentMapping> mapping = message.getMappingsList();
+
+        // insert all of the mappings
+        for (Sync.CommentMapping entry: mapping) {
+            db.insertKnownPost(entry.getPostHash(), entry.getCommentHash());
+        }
+    }
+
+    public void syncData(Sync.HashData message) {
```

Comment:
Add javadoc comment about the functions of this method.

Reply:
Fixed in commit 6d83384